### PR TITLE
Add bloc to sign up

### LINKED ISSUE #15: [BLoC] Sign up
**Branch:** _feature/sign-up-bloc_

### PR DESCRIPTION
**Pull request by taruuuch**
Fix #15

### CODE REVIEW

File: lib/pages/sign_in/widgets/signin_form.dart [hunk 1]
```diff
@@ -21,7 +21,7 @@ class _SignInFormState extends State<SignInForm> {
     final SignInBloc bloc = BlocProvider.of<SignInBloc>(context);
     return new Form(
       key: _formKey,
-        child: new Padding(
+      child: new Padding(
       padding: const EdgeInsets.symmetric(vertical: 0.0, horizontal: 15.0),
```
Comment (stenvix) [inline lib/pages/sign_in/widgets/signin_form.dart:25]: Move padding values to one place (Like AppStyles or just Styles class)

File: lib/pages/sign_up/widgets/signup_bloc.dart [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  }
+
+  Stream<SignUpState> _signUp({String email, String password}) async* {
+    yield SignUpState.loading();
+
+    try {
+      await UserProvider().signUp(email, password);
+      yield SignUpState.success();
+    } catch (e) {
+      yield SignUpState.error('sign up error');
```
Comment (stenvix) [inline lib/pages/sign_up/widgets/signup_bloc.dart]: Print errors from server (like if email is already in use)

File: lib/pages/sign_up/widgets/signup_form.dart [hunk 3]
```diff
@@ -33,12 +54,19 @@ class SignUpFormState extends State<SignUpForm> {
...
+              controller: _passwordController,
+              decoration: InputDecoration(
+                labelText: 'Enter password',
+                border: new OutlineInputBorder(
+                    borderSide: new BorderSide(color: Colors.grey)),
+              ),
               obscureText: true,
               keyboardType: TextInputType.text,
               validator: (value) {
                 if (value.isEmpty) {
```
Comment (stenvix) [inline lib/pages/sign_up/widgets/signup_form.dart:66]: Move form validators to one place (Like FormValidators class)

**Review by stenvix — CHANGES_REQUESTED** [review]
Need some improvements, read comments please

File: lib/pages/sign_up/widgets/signup_form.dart (hunk 3 above)
Comment (taruuuch) [reply lib/pages/sign_up/widgets/signup_form.dart:66]: For this feature needs one more issue

File: lib/pages/sign_in/widgets/signin_form.dart (hunk 1 above)
Comment (taruuuch) [reply lib/pages/sign_in/widgets/signin_form.dart:25]: For this feature needs one more issue

Comment (taruuuch) [reply lib/pages/sign_in/widgets/signin_form.dart:25]: Fix in #21

File: lib/pages/sign_up/widgets/signup_form.dart (hunk 3 above)
Comment (taruuuch) [reply lib/pages/sign_up/widgets/signup_form.dart:66]: Fix in #22

**Pull request merged**